src/main/java/io/vertx/mqtt/impl/MqttClientImpl.java

```diff
@@ -538,6 +587,9 @@ void publishRelease(int publishMessageId) {
 
     io.netty.handler.codec.mqtt.MqttMessage pubrel = MqttMessageFactory.newMessage(fixedHeader, variableHeader, null);
 
+    // we need to remove here because we want to add the element to the end of list
+    qos2outbound.remove(publishMessageId);
```

Comment:
not sure we need to change the order, putting the PUBREL at the end of the queue